### ISSUE TITLE
Questions about code

### ISSUE THREAD

**Issue by yutongred** [CR-Gjx/LTF-Label-Transformation-Framework#2]
Hi authors
Interesting work! But I have a few questions about the code

1. The code seems a bit messy, (some unused defines)
     Also, only the 'f-m' implemention is correct?
2. In your zip file, your preTrain process .npy  data seems not included
     Do I need to run the preTrain?
3. Could you release the continuous label shift codes?
4. confusing implementation:
       a) In your paper, you said LT takes in noise and source label, but in your code you just use random label?
       b) a bit confused about the reinforce loss, could you explain a little ?
            for example, why use this and what is 0.15 in the code

**Comment by CR-Gjx** [comment]
Thanks for your suggestions:
1. Actually, there are some unused defines in my code, and I will further clean my code.
2. Normally, the preTrain option is for the training of generator (P(X|Y)) in our paper. I will re-upload a new version of code to eliminate the bugs.
3. I will upload it as soon as possible. You know,  there are many messy code blocks in my repository.
4. (a)Adding noise into the LT is for the randomness of label transformation, while the randomness is reflected by the final sampling of label transformation (Refer to the 275 line of main.py).   (b) Reinforce loss is one method to approximate the target distribution (Please refer to [https://stillbreeze.github.io/REINFORCE-vs-Reparameterization-trick/](url) ). However, the Reinforce trick suffers from the high variance problem, so we reduce  0.15 to reduce its variance. This is a normal trick for Reinforce algorithm.

**Comment by yutongred** [comment]
Thanks for your reply, been busy with exams last few days. :)
I will read the code and reference ,hope will solve my questions!

**Comment by CR-Gjx** [comment]
I have uploaded the regression synthetic experiment. Please run the experiments referring to the ReadMe.
In addition, you can use the original training dataset in the classification experiment to avoid loading preTrain .npy data

**Comment by CR-Gjx** [comment]
Thanks for your attention, please contact me if you have further questions.
